File: solutions/prospective/algorithms/patterns/js/two_pointers.js
/**
 * Two Pointers Template
 * 
 * Common pattern for sorted arrays (e.g., 2Sum II) or checking palindromes.
 * - Left starts at 0, Right starts at len-1.
 * - Move pointers based on condition.
 */
// === MEMO START ===
function twoPointers(arr, target) {
    let l = 0;
    let r = arr.length - 1;
    
    while (l < r) {
        const currentSum = arr[l] + arr[r];
        if (currentSum === target) {
            return [l, r];
        } else if (currentSum < target) {
            l++;
        } else {
            r--;
        }
    }
    return [];
}

File: solutions/prospective/algorithms/patterns/lua/two_pointers.lua
--[[
Two Pointers Template

Common pattern for sorted arrays (e.g., 2Sum II) or checking palindromes.
- Left starts at 1, Right starts at #arr.
- Move pointers based on condition.
--]]
-- === MEMO START ===
function two_pointers(arr, target)
    local l = 1
    local r = #arr
    
    while l < r do
        local current_sum = arr[l] + arr[r]
        if current_sum == target then
            return {l, r}
        elseif current_sum < target then
            l = l + 1
        else
            r = r - 1
        end
    end
    return {}
end

